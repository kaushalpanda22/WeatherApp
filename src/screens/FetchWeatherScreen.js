import React, { useCallback, useEffect } from 'react';
import { View } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import WeatherInfo from '../components/weatherInfo';
import styles from '../constants/styles';
import * as weatherActions from '../store/actions/search';
import NetworkUtils from '../utils/NetworkUtil';

const FetchWeatherScreen = props => {
    const { weatherData, errorMessage, isLoading } = useSelector(state => state.search);
    const dispatch = useDispatch();
    const pincode = props.navigation.getParam('param_pincode');

    useEffect(() => {
        fetchData()
    }, [dispatch]);

    const fetchData = useCallback(async () => {
        const isConnected = await NetworkUtils.isNetworkAvailable()
        dispatch(isConnected ?
            weatherActions.searchByPincode(pincode) : weatherActions.getDataFromLocalStorage(pincode));
    }, [dispatch])

    return (
        <View style={styles.container}>
            <WeatherInfo
                weatherData={weatherData}
                errorMessage={errorMessage}
                isLoading={isLoading}
            />
        </View>
    );
}

FetchWeatherScreen.navigationOptions = () => {
    return {
        title: 'Weather Forcast',
    }
}

export default FetchWeatherScreen;