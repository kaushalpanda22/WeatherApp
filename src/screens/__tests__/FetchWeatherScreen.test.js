/**
 * @format
 */

import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme'
import FetchWeatherScreen from '../FetchWeatherScreen';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';


let wrapper;
const mockStore = configureMockStore([thunk]);
const store = mockStore({
    search: {
        weatherData: {},
        errorMessage: '',
        isLoading: false
    }
});

wrapper = shallow(
    <Provider store={store}>
        <FetchWeatherScreen navigation={{ getParam: jest.fn() }} />
    </Provider>
);

it('FetchWeatherScreen Screen renders correctly', () => {
    renderer.create(<Provider store={store}>
        <FetchWeatherScreen navigation={{ getParam: jest.fn() }} />
    </Provider>).toJSON();
});

describe('FetchWeatherScreen', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            expect(wrapper).toMatchSnapshot()
        });
    });
});

test('navigation options', () => {
    const naviProp = { navigation: { navigate: () => { } } };
    const navigationOptions = FetchWeatherScreen.navigationOptions(naviProp);

    expect(navigationOptions).toMatchSnapshot();
});
