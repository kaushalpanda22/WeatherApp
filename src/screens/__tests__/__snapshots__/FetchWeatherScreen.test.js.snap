// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchWeatherScreen Rendering should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`navigation options 1`] = `
Object {
  "title": "Weather Forcast",
}
`;
